File: android/app/src/main/java/com/thirtysevendegrees/app/designsystem/organisms/EmptyStateView.kt
package com.thirtysevendegrees.app.designsystem.organisms

import androidx.compose.foundation.layout.*
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Inbox
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.thirtysevendegrees.app.designsystem.atoms.AppButton
import com.thirtysevendegrees.app.designsystem.atoms.ButtonSize
import com.thirtysevendegrees.app.designsystem.atoms.ButtonVariant
import com.thirtysevendegrees.app.designsystem.tokens.*

@Composable
fun EmptyStateView(
    title: String,
    subtitle: String,
    actionTitle: String? = null,
    onAction: (() -> Unit)? = null,
    modifier: Modifier = Modifier
) {
    Column(
        modifier = modifier
            .fillMaxWidth()
            .padding(horizontal = Spacing.XXL),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Spacer(modifier = Modifier.height(Spacing.XXXL))
        Icon(
            imageVector = Icons.Default.Inbox,
            contentDescription = null,
            modifier = Modifier.size(IconSize.XXL),
            tint = AppColors.textTertiaryLight
        )
        Spacer(modifier = Modifier.height(Spacing.MD))
        Text(title, style = MaterialTheme.typography.headlineMedium, color = AppColors.textSecondaryLight)
        Spacer(modifier = Modifier.height(Spacing.XS))
        Text(subtitle, style = MaterialTheme.typography.bodySmall, color = AppColors.textTertiaryLight)
        if (actionTitle != null && onAction != null) {
            Spacer(modifier = Modifier.height(Spacing.LG))
            AppButton(
                text = actionTitle,
                onClick = onAction,
                variant = ButtonVariant.Secondary,
                size = ButtonSize.Medium,
                modifier = Modifier.width(120.dp)
            )
        }
    }
}

File: ios/ThirtySevenDegrees/DesignSystem/Organisms/EmptyStateView.swift
import SwiftUI

struct EmptyStateView: View {
    let title: String
    let subtitle: String
    var actionTitle: String? = nil
    var action: (() -> Void)? = nil

    var body: some View {
        VStack(spacing: .spaceMD) {
            Spacer().frame(height: .spaceXXXL)

            Image(systemName: "tray")
                .font(.system(size: IconSize.xxl.rawValue))
                .foregroundStyle(.textTertiary)

            Text(title)
                .font(.headline)
                .foregroundStyle(.textSecondary)

            Text(subtitle)
                .font(.bodySmall)
                .foregroundStyle(.textTertiary)
                .multilineTextAlignment(.center)

            if let actionTitle, let action {
                AppButton(title: actionTitle, variant: .secondary, size: .medium, action: action)
                    .frame(width: 120)
            }
        }
        .frame(maxWidth: .infinity)
        .padding(.horizontal, .spaceXXL)
    }
}
